client.go

```diff
@@ -276,6 +296,29 @@ func NewClient(appID string, appKey string, host string, port int, realm string,
 	return c, nil
 }
 
+// parseError :
+//  non-exportable helper to parse error response codes between handled (loosely expected http status)
+//  and non-handled status codes.
+func parseError(response *http.Response) (*HttpError, error) {
+	httpError := new(HttpError)
+	handledStatus := []int{400, 404, 500}
```

Comment:
Would we want to consider any 4xx or 5xx an error? I could see IdP returning a 503, for instance.